src/WebJobs.Script.WebHost/Diagnostics/DiagnosticEventTableStorageRepository.cs

```diff
@@ -53,16 +53,16 @@ internal CloudTableClient TableClient
                 if (!_environment.IsPlaceholderModeEnabled() && _tableClient == null)
                 {
                     string storageConnectionString = _configuration.GetWebJobsConnectionString(ConnectionStringNames.Storage);
-                    if (string.IsNullOrEmpty(storageConnectionString))
-                    {
-                        _logger.LogError("Azure Storage connection string is empty or invalid. Unable to write diagnostic events.");
-                    }
-
-                    if (CloudStorageAccount.TryParse(storageConnectionString, out CloudStorageAccount account))
+                    if (!string.IsNullOrEmpty(storageConnectionString)
+                        && CloudStorageAccount.TryParse(storageConnectionString, out CloudStorageAccount account))
                     {
                         var tableClientConfig = new TableClientConfiguration();
                         _tableClient = new CloudTableClient(account.TableStorageUri, account.Credentials, tableClientConfig);
```

Comment:
If more than one thread accesses "TableClient" internal property, wouldn't that cause this code to create multiple instances of `CloudTableClient`? and when it's time to do the flush, we may do that for the last one only. Should we make sure this code executes only for one thread ?

Reply:
In case there are 2 threads trying to create a tableClient at same time, They would end up creating a tableClient pointing to the same table. As a result, the logs would end up in the same table. As of right now only flush logs is using tableClient, which runs once every 10 minutes.

Reply:
`_tableClient = new CloudTableClient`  - everytime we executes this , we are creating a new `CloudTableClient` intance. So if 2 threads ends up executing (line 60), one after the other, it is possible for the `TableClient` property to return 2 different objects.

Reply:
right, but both objects would end up pointing to the same table in azure storage. As the name is based on the date. So even if they end up using different objects the logs would end up in the same table.